### ISSUE TITLE
Ensemble question

### ISSUE THREAD

**Issue by homelifes** [ruotianluo/self-critical.pytorch#139]
Hi,
May I ask about the compuation of Ensemble in AttEnsemble.py
We are supposed to take multiple models and run them. Then we take the softmax out for each model (or their logs if using beam search) and add them together and divide them by the number of models (averaging all). So for example if we have 2 models, we would do: (softmax1+softmax2)/2 and then take the maximum output to get the word prediction of the ensemble. Am I correct? 
If yes, in your code you are doing:
        `logprobs = torch.stack([F.softmax(m.logit(output[i]), dim=1) for i,m in enumerate(self.models)], 2).mul(self.weights).div(self.weights.sum()).sum(-1).log()
`
You are stacking the models along the vocab dimension and then diving them by the number of models and then summing the vocab dimension by using sum(-1), so your getting a single value for all the model outputs. May you please explain? Thanks

**Comment by homelifes** [comment]
@ruotianluo

**Comment by ruotianluo** [comment]
Sorry, because I stack at dim2
